scene/2d/cpu_particles_2d.cpp

```diff
@@ -53,9 +53,11 @@ void CPUParticles2D::set_amount(int p_amount) {
 	{
 		PoolVector<Particle>::Write w = particles.write();
 
-		for (int i = 0; i < p_amount; i++) {
-			w[i].active = false;
-		}
+		// each particle must be set to false
+		// zeroing the data also prevents uninitialized memory being sent to GPU
+		zeromem(static_cast<void*>(&w[0]), p_amount * sizeof(Particle));
+		// cast to prevent compiler warning .. note this relies on Particle not containing any complex types.
+		// an alternative is to use some zero method per item but the generated code will be far less efficient.
```

Comment:
Suggestions welcome on a more compliant way of doing this that is just as efficient (e.g. std::fill_n)...

Reply:
That's not pretty :D But if it does the trick, that should be fine.
Maybe add a comment near `struct Particle` in the header to warn against adding non-trivial types, since I guess this wouldn't fail compiling but just give us another nice heisenbug ;)

Reply:
Yeah that's a good idea, i'll add that. :)